--- lib/Makefile.in.orig	2016-11-16 17:14:54.060376848 +0000
+++ lib/Makefile.in	2016-11-16 17:15:12.701375842 +0000
@@ -272,7 +272,7 @@
 my_pkglibdir = $(pkglibdir)
 my_pkglib_DATA = rancid.pm acos.pm aeos.pm arbor.pm bigip.pm ciscowlc.pm \
 		 dell.pm edgemax.pm edgerouter.pm eos.pm exos.pm foundry.pm \
-		 ios.pm iosxr.pm junos.pm mrv.pm nxos.pm panos.pm rbt.pm \
+		 ios.pm iosxr.pm junos.pm mos.pm mrv.pm nxos.pm panos.pm rbt.pm \
 		 wavesvros.pm
 
 pkglib_in = $(my_pkglib_DATA:%=%.in)
@@ -539,6 +539,7 @@
 iosxr.pm: iosxr.pm.in
 junos.pm: junos.pm.in
 rancid.pm: rancid.pm.in
+mos.pm: mos.pm.in
 mrv.pm: mrv.pm.in
 nxos.pm: nxos.pm.in
 panos.pm: panos.pm.in
--- lib/Makefile.am.orig	2016-11-16 17:15:50.557373799 +0000
+++ lib/Makefile.am	2016-11-16 17:16:15.309372463 +0000
@@ -9,7 +9,7 @@
 my_pkglibdir = $(pkglibdir)
 my_pkglib_DATA = rancid.pm acos.pm aeos.pm arbor.pm bigip.pm ciscowlc.pm \
 		 dell.pm edgemax.pm edgerouter.pm eos.pm exos.pm foundry.pm \
-		 ios.pm iosxr.pm junos.pm mrv.pm nxos.pm panos.pm rbt.pm \
+		 ios.pm iosxr.pm junos.pm mos.pm mrv.pm nxos.pm panos.pm rbt.pm \
 		 wavesvros.pm
 pkglib_in = $(my_pkglib_DATA:%=%.in)
 
@@ -69,6 +69,7 @@
 iosxr.pm: iosxr.pm.in
 junos.pm: junos.pm.in
 rancid.pm: rancid.pm.in
+mos.pm: mos.pm.in
 mrv.pm: mrv.pm.in
 nxos.pm: nxos.pm.in
 panos.pm: panos.pm.in
--- etc/rancid.types.base.orig	2016-11-16 21:20:57.327579992 +0000
+++ etc/rancid.types.base	2016-11-16 21:22:54.109573689 +0000
@@ -469,6 +469,16 @@
 juniper-srx;command;junos::ShowConfiguration;show configuration system
 juniper-srx;command;junos::ShowConfiguration;show configuration vlans
 #
+metamako;script;rancid -t metamako
+metamako;login;clogin
+metamako;module;mos
+metamako;inloop;mos::inloop
+metamako;command;mos::ShowVersion;show version
+metamako;command;mos::ShowEnv;show env all
+metamako;command;mos::ShowBoot;show boot
+metamako;command;mos::ShowInventory;show inventory
+metamako;command;mos::WriteTerm;show running-config
+#
 mikrotik;script;mtrancid
 mikrotik;login;mtlogin
 #
